logstreams/src/main/java/io/zeebe/logstreams/storage/atomix/AtomixLogStorage.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.logstreams.storage.atomix;
+
+import io.atomix.storage.journal.Indexed;
+import io.zeebe.logstreams.spi.LogStorage;
+import io.zeebe.logstreams.spi.LogStorageReader;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.NoSuchElementException;
+import java.util.concurrent.CompletableFuture;
+
+public class AtomixLogStorage implements LogStorage {
+  private final AtomixReaderFactory readerFactory;
+  private final AtomixLogCompactor logCompacter;
+  private final AtomixAppenderSupplier appenderSupplier;
+
+  private boolean opened;
+
+  public AtomixLogStorage(
+      final AtomixReaderFactory readerFactory,
+      final AtomixLogCompactor logCompacter,
+      final AtomixAppenderSupplier appenderSupplier) {
+    this.readerFactory = readerFactory;
+    this.logCompacter = logCompacter;
+    this.appenderSupplier = appenderSupplier;
+  }
+
+  @Override
+  public LogStorageReader newReader() {
+    return new AtomixLogStorageReader(readerFactory.create());
+  }
+
+  @Override
+  public long append(final ByteBuffer blockBuffer) throws IOException {
+    throw new UnsupportedOperationException("Not supported");
```

Comment:
appendAsync().join() could be called right?

Reply:
Can now be removed